pilot/pkg/xds/ads.go

```diff
@@ -323,6 +327,92 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 	}
 }
 
+// StreamSecrets serves SDS discovery requests and SDS push requests
+func (s *DiscoveryServer) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecretsServer) error {
```

Comment:
Ah, this is unimplemented yet

Reply:
We only need one of the methods for SDS - I used exactly what istio-agent SDS server is using.